dialogue-futures/src/test/java/com/palantir/dialogue/futures/DialogueFuturesTest.java

```diff
@@ -0,0 +1,216 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.futures;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import com.google.common.util.concurrent.Futures;
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.SettableFuture;
+import java.util.concurrent.CancellationException;
+import org.junit.jupiter.api.Test;
+
+class DialogueFuturesTest {
+
+    @Test
+    void testTransform_success() throws Exception {
+        SettableFuture<String> original = SettableFuture.create();
+        ListenableFuture<String> doubled = DialogueFutures.transform(original, value -> value + value);
+        original.set("a");
+        assertThat(doubled.get()).isEqualTo("aa");
+    }
+
+    @Test
+    void testTransform_originalCancel() {
+        SettableFuture<String> original = SettableFuture.create();
+        ListenableFuture<String> doubled = DialogueFutures.transform(original, value -> value + value);
+        assertThat(doubled.cancel(false)).isTrue();
```

Comment:
think this should be `original.cancel(false)` right? (otherwise it's identical to the one below)